contracts/YearnV2Strat.sol

```diff
@@ -0,0 +1,145 @@
+            }
+            yToken.withdraw(sharesForAmount(missingAmount) + 1, address(this));  // +1 is a fix for a rounding issue
+        }
+        underlying.transfer(address(vault), amount);
+        if(!zeroUnutilized) syncUntilized(); // unutilized can only go down when withdrawing.
+    }
+
+    function totalYearnDeposits() public view returns (uint) {
+        return yToken.balanceOf(address(this))
+                .mul(yToken.pricePerShare())
+                .div(10**18); // TODO: Dynamic precision
+    }
+
+    function calcTotalValue() external view override returns (uint) {
+        return Math.max(totalYearnDeposits(), 1) // cannot be lower than 1 because we subtract 1 after
+        .sub(1) // account for dust
+        .add(underlying.balanceOf(address(this)));
+    }
+
+    // IMPORTANT: This function can only be called by the timelock to recover any token amount including deposited yToken and underlying
+    // However, the owner of the timelock must first submit their request and wait 2 days before confirming.
+    // This gives depositors a good window to withdraw before a potentially malicious rescue
+    // The intent is for the owner to be able to rescue funds in the case they become stuck after launch
+    // However, users should not trust the owner and watch the timelock contract least once a week on Etherscan
+    // In the future, the timelock contract will be destroyed and the functionality will be removed after the code gets audited
+    function rescue(address _token, address _to, uint _amount) external {
+        require(msg.sender == address(timelock));
+        IERC20(_token).transfer(_to, _amount);
+    }
+
+    // Any tokens (other than the yToken and underlying) that are sent here by mistake are recoverable by the vault owner
+    function sweep(address _token, address _to) public onlyOwner {
+        require(_token != address(yToken) && _token != address(underlying));
+        IERC20(_token).transfer(_to, IERC20(_token).balanceOf(address(this)));
+    }
+
+    function pauseDeposits(bool value) public onlyOwner {
+        depositsPaused = value;
+    }
+
+    // Bypasses withdrawal cap. Should be used with care. Can cause Yearn slippage with large amounts.
+    function withdrawShares(uint shares) public onlyOwner {
+        yToken.withdraw(shares, address(this));
+        syncUntilized();
+    }
+
+    // Bypasses withdrawal cap. Should be used with care. Can cause Yearn slippage with large amounts.
+    function withdrawUnderlying(uint amount) public onlyOwner {
+        yToken.withdraw(sharesForAmount(amount), address(this));
+        syncUntilized();
+    }
+
+    // Bypasses withdrawal cap. Should be used with care. Can cause Yearn slippage with large amounts.
+    function withdrawAll() public onlyOwner {
+        yToken.withdraw(yToken.balanceOf(address(this)), address(this));
```

Comment:
these can be simplified:

withdraw() <- withdraw all to sender address

withdraw(amount) <- withdraw amount to sender address

withdraw(amount,to) <- withdraw amount to 'to' address

Reply:
Ah I'm not familiar with Vyper so I didn't know the syntax for optional parameters. Will add overloaded functions to the interface and use them.

Reply:
https://github.com/InverseFinance/inverse-protocol/pull/1/commits/92a512ff99690b8cb828809aa18053e2ad0a468f